src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/DynamicFrameworkContract.kt

```diff
@@ -32,10 +33,10 @@ data class DynamicFrameworkContract(
   @Embedded
   val contractEligibility: ContractEligibility,
 
-  @ManyToOne @JoinColumn(name = "nps_region_id")
+  @ManyToOne(fetch = FetchType.LAZY) @JoinColumn(name = "nps_region_id")
   val npsRegion: NPSRegion? = null,
 
-  @ManyToOne @JoinColumn(name = "pcc_region_id")
+  @ManyToOne(fetch = FetchType.LAZY) @JoinColumn(name = "pcc_region_id")
   val pccRegion: PCCRegion? = null,
```

Comment:
I'm not sure we want lazy loading.  The current service calls to get interventions and contracts are immediately accessing the service category/provider/nps and pcc regions and manipulating them into DTOs - so wouldn't the lazy loading just result in multiple calls to the database rather than just one eager load?
I can understand for a oneToMany - but these are small objects.

Reply:
yeh good point. i think you're right that we will likely always want to access the service provider and service category information when returning results about interventions. for regions i'm not sure.  our existing endpoints do not pull region data. but i also agree they're small objects. will pull these changes out for now.